x/ecocredit/server/core/features/msg_remove_class_creator.feature

```diff
@@ -0,0 +1,124 @@
+Feature: Msg/RemoveClassCreator
+
+  A class creator can be removed:
+  - when the class creator is exist
+  - when the authority is a governance account address
+  - the class creator is removed
+
+  Rule: The class creator is exist
+
+    Scenario: The class creator is exist
+      Given class creators with properties
+      """
+      {
+        "creators":[
+            "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+        ]
+      }
+      """
+      When alice attempts to remove a class creator with properties
+      """
+      {
+        "authority":"regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68",
+        "creator": "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+      }
+      """
+      Then expect no error
+  
+    Scenario: The class creator does not exist
+      Given class creators with properties
+      """
+      {
+        "creators":[
+            "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+        ]
+      }
+      """
+      When alice attempts to remove a class creator with properties
+      """
+      {
+        "authority":"regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68",
+        "creator": "regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68"
+      }
+      """
+      Then expect the error "class creator regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68: not found"
+
+  Rule: The authority must be governance account address
+
+    Scenario: The authority is a governance account address
+      Given class creators with properties
+      """
+      {
+        "creators":[
+            "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+        ]
+      }
+      """
+      When alice attempts to remove a class creator with properties
+      """
+      {
+        "authority":"regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68",
+        "creator": "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+      }
+      """
+      Then expect no error
+
+    Scenario: The authority is not a governance account address
+      When alice attempts to remove a class creator with properties
+      """
+      {
+        "authority":"regen1fua8speyxgempgy06gpfs0p4z32zznkqakm57s",
+        "creator":"regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+      }
+      """
+      Then expect error contains "expected gov account as only signer for proposal message"
+
+  Rule: The class creator is removed
+
+    Scenario: The class creator is removed
+    Given class creators with properties
+      """
+      {
+        "creators":[
+            "regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+        ]
+      }
+      """
+      When alice attempts to remove a class creator with properties
+      """
+      {
+        "authority":"regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68",
+        "creator":"regen156d26rl52y3wl865pr5x9q2vqetuw9kf0642sa"
+      }
+      """
+      Then expect no error
+      And expect class creators list to be empty
```

Comment:
We check the error and the outcome in separate scenarios so we only need to check the list here. We already check the error in the cases above and the goal is to have each scenario focused on testing one outcome (i.e. if there is an error should be tested separately from what the results are) - at least for unit tests.

```suggestion
      Then expect class creators list to be empty
```